24_find_lab/findlab.txt

```diff
@@ -1,24 +1,44 @@
 1. Write command lines to find using "find":
 
   -. All files owned by yourself in /tmp folder
+With ls -l to show permissions as well...
+
+find ~/tmp -type f -user $USER -exec ls -l {} \;
 
   -. All files and directories not owned by you in /tmp folder
 
+find ~/tmp ! -user $USER -exec ls -l {} \;
+
+
   -. All files, folders and links in your home folder whose name ends with a ".c"
 
+find . -type f -name '*.c'
+
   -. All files in your home folder whose name starts with an "a", "e", "i", "o" or "u"
 
+find ~/ -type f-name '[aeiou]*'
+
   -. All files in your home directory whose size is above 10MB
 
+find ~/ -type f -size +10M
+
   -. All files in your home directory which were modified in the past 2 days
 
+find ~/ -type f -mtime -2 -exec ls -l {} \;
+
   -. All files in your home directory which were not modified for over a month, and size is above 10MB.
 
+find ~/ -mtime +30 -size +10M
+
 2. Create a hard link to a file of your choice and put it in a subdirectory under your home folder. How can you use "find" to locate the hard link given the original file?
 
+ln $(realpath myfile.txt) ~/lab/linkMyfile
+find ~/ -type f -samefile ~/myfile.txt
+
 3. Create 3 empty text files a.txt, b.txt and c.txt. Use a single find command to:
   - copy a.txt to a.txt.backup
   - copy b.txt to b.txt.backup
   - copy c.txt to c.txt.backup
 What was the command?  
+find -name '[a-c].txt' -exec cp {}  {}.backupm \;
```

Comment:
Very Good!. Do note that on SOME versions of unix you'll need to add quotes around {} i.e. `find . -name '[a-c].txt' -exec cp {} "{}".backupm \;`